DotNetHowTo/DotNetHowTo/Hotel.cs

```diff
@@ -33,6 +33,12 @@ public partial class Hotel
         [IsFilterable, IsSortable, IsFacetable]
         public bool? ParkingIncluded { get; set; }
 
+        // SmokingAllowed reflects whether any room in the hotel allows smoking.
+        // The JsonIgnore attribute indicates that a field should not be created 
+        // in the index for this property and it will only be used by code in the client.
+        [JsonIgnore]
+        public bool? SmokingAllowed => (Rooms != null) ? Array.Exists(Rooms, element => element.SmokingAllowed == true) : (bool?)null;
```

Comment:
There's no need to make this nullable, since it's not actually stored. The other properties are nullable because any field in an Azure Search index can be null. Outside of that scenario, nullable primitives should be avoided because it just puts extra burden on the caller to check for null.

I would write it like this:

```csharp
public bool SmokingAllowed => Rooms?.Any(r => r.SmokingAllowed == true) ?? false;
```

Reply:
This is currently a projection of the underlying value, meaning if all SmokingAllowed in Rooms are null or Rooms itself is null, this would reflect null.  Seems more consistent to me to actually be null since people have to look for null on every other value.  I'd be more inclined to agree if there wasn't such a close tie between an index field value and this one.

Reply:
Fair enough, if null means "we don't know whether smoking is allowed or not"